src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -58,9 +58,14 @@ internal ItemManager(RecentItemsManager recentItemsManager, IGlobalOptionService
         private static SegmentedList<VSCompletionItem> SortCompletionitems(AsyncCompletionSessionInitialDataSnapshot data, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
-            var items = new SegmentedList<VSCompletionItem>(data.InitialItemList);
-            items.Sort(VSItemComparer.Instance);
```

Comment:
Previously we relied on the sort to ensure all VSCompletionItem contains a Roslyn CompletionItem (which is what subsequent selection, sorting, filtering operations are using.) The bug is caused by non-roslyn IAsyncCompletionSource (e.g. intellicode) triggers and returns a single item (w/o other sources involved), and `SegmentList.Sort` is a no-op with a single element. We have observed an occurrence of this bug in one of the watson dumps.